src/extension.ts

```diff
@@ -104,6 +104,24 @@ function getClientConfig(context: ExtensionContext) {
     return false;
   }
 
+  function resolveVariablesInString(value: string) {
+    return value.replace('${workspaceFolder}', workspace.rootPath);
+  }
+
+  function resloveVariablesInArray(value: any[]) {
+    return value.map(v => resolveVariables(v));
+  }
+
+  function resolveVariables(value: any) {
```

Comment:
Can you make the `any` be `string`? Or do you get type errors?

Reply:
resolveVariables() is called for all options from configMapping. There are, for example, `cquery.misc.enableIndexing`, which is a boolean, `cquery.misc.indexerCount` - number (as you can see, the function will leave them unmodified). Also there are string arrays (`cquery.launch.args`), I want `${workspaceFolder}` to be resolved in array elements.
I do not see how it could work with only `string` argument type.